include/posix/pthread.h

```diff
@@ -157,6 +157,20 @@ static inline int pthread_condattr_destroy(pthread_condattr_t *att)
 		.sem = &name##_psem,		\
 	}
 
+/**
+ * @brief Declare a pthread sem
+ *
+ * Declaration API for a pthread sem.  This is not a POSIX API, it's
+ * provided to better conform with Zephyr's allocation strategies for
+ * kernel objects.
+ *
+ * @param name Symbol name of the sem
+ */
+#define PTHREAD_SEM_DEFINE(name, initial_count, count_limit)	\
```

Comment:
lets not do that, you can also define the semaphore dynamically. Not sure what you are trying to conform to. Define SEM_VALUE_MAX somewhere and make it UINT_MAX and use this for the max count, this is how it is done in posix.